classes/oai/ojs/OAIDAO.inc.php

```diff
@@ -252,72 +242,64 @@ public function _getRecordsRecordSet($setIds, $from, $until, $set, $submissionId
-				NULL AS issue_id,
-				dot.tombstone_id,
-				dot.set_spec,
-				dot.oai_identifier
-			FROM	data_object_tombstones dot' . '
-				' . (isset($journalId) ? 'JOIN data_object_tombstone_oai_set_objects tsoj ON (tsoj.tombstone_id = dot.tombstone_id AND tsoj.assoc_type = ' . ASSOC_TYPE_JOURNAL . ' AND tsoj.assoc_id = ?)' : '') . '
-				' . (isset($sectionId) ? 'JOIN data_object_tombstone_oai_set_objects tsos ON (tsos.tombstone_id = dot.tombstone_id AND tsos.assoc_type = ' . ASSOC_TYPE_SECTION . ' AND tsos.assoc_id = ?)' : '') . '
-			WHERE	1=1
-				' . (isset($set) ? ' AND (dot.set_spec = ? OR dot.set_spec LIKE ?)' : '') . '
-				' . ($from ? ' AND dot.date_deleted >= ' . $this->datetimeToDB($from) : '') . '
-				' . ($until ? ' AND dot.date_deleted <= ' . $this->datetimeToDB($until) : '') . '
-				' . ($submissionId ? ' AND dot.data_object_id = ?' : '') . '
-			ORDER BY ' . $orderBy,
-            $params
-        );
+        $submissionsQuery = DB::table('submissions AS a')
+            ->select(DB::raw('GREATEST(a.last_modified, i.last_modified) AS last_modified,
+                                a.submission_id AS submission_id,
+                                j.journal_id AS journal_id,
+                                s.section_id AS section_id,
+                                i.issue_id,
+                                NULL AS tombstone_id,
+                                NULL AS set_spec,
+                                NULL AS oai_identifier'))
+            ->join('publications AS p', 'a.current_publication_id', '=', 'p.publication_id')
+            ->join('publication_settings AS psissue', function($join) {
+                $join->on('psissue.publication_id', '=', 'p.publication_id');
+                $join->where('psissue.setting_name', '=', DB::raw('\'issueId\''));
+                $join->where('psissue.locale', '=', DB::raw('\'\''));
+            })
+            ->join('issues AS i', function($join) {
+                $join->on(DB::raw('CAST(i.issue_id AS CHAR(20))'), '=', 'psissue.setting_value');
+            })
+            ->join('sections AS s', 's.section_id', '=', 'p.section_id')
+            ->join('journals AS j', 'j.journal_id', '=', 'a.context_id')
+            ->where('i.published', '=', 1)
+            ->where('j.enabled', '=', 1)
+            ->where('a.status', '=', PKPSubmission::STATUS_PUBLISHED);
+        if ($excludeJournals) $submissionsQuery->whereNotIn('j.journal_id', $excludeJournals);
+        if (isset($journalId)) $submissionsQuery->where('j.journal_id', '=', (int) $journalId);
+        if (isset($sectionId)) $submissionsQuery->where('p.section_id', '=', (int) $sectionId);
+        if ($from) $submissionsQuery->where('GREATEST(a.last_modified, i.last_modified)', '>=', $from);
+        if ($until) $submissionsQuery->where('GREATEST(a.last_modified, i.last_modified)', '<=', $until);
+        if ($submissionId) $submissionsQuery->where('a.submission_id', '=', (int) $submissionId);
+
```

Comment:
According to the documentation we could maybe use the conditional clauses (s. https://laravel.com/docs/7.x/queries#conditional-clauses) instead of `if ...`, or?
If so, the result is surely the same, so I will let you decide what to use :-)

Reply:
The conditional clauses are a little more verbose, but using them permits me to build the query in one (long) statement, and I wonder if eventually we'll be able to benefit from this in other ways. In any case, I made the change.